src/pages/why-do-we-write-super-props/index.zh-hant.md

```diff
@@ -0,0 +1,193 @@
+---
+title: 為什麼我們要寫 super(props)?
+date: '2018-11-30'
+spoiler: 在結尾有一個轉折。
+---
+
+
+我聽說 [Hooks](https://reactjs.org/docs/hooks-intro.html) 正夯， 但我想要從揭開 *class* 有趣的面紗來開始這個部落格。這樣如何！
+
+**這些解釋對於有效地使用 React 並*不*重要，但對於喜歡深入了解其運作的你，將會發現當中有趣的事情。**
+
+---
+
+首先，我生命中寫過 `super(props)` 的次數比我想像的還多：
+
+```jsx{3}
+class Checkbox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isOn: true };
+  }
+  // ...
+}
+```
+
+沒錯，[類別欄位提案](https://github.com/tc39/proposal-class-fields) 讓我們能省略宣告：
+
+```jsx
+class Checkbox extends React.Component {
+  state = { isOn: true };
+  // ...
+}
+```
+
+在 2015 年， React 0.13 新增對一般類別的支援，[計劃](https://reactjs.org/blog/2015/01/27/react-v0.13.0-beta-1.html#es7-property-initializers) 藉由定義 `constructor` 和呼叫 `super(props)` 的語法當作臨時的解決方法，直到類別欄位提供合適的替代方案。
```

Comment:
```suggestion
類似這樣的語法是在 2015 年React 0.13 新增對一般類別的支援時就有的[計劃](https://reactjs.org/blog/2015/01/27/react-v0.13.0-beta-1.html#es7-property-initializers) 。
。定義 `constructor` 和呼叫 `super(props)` 的語法一直都被當作是暫時的解決方法，直到類別欄位提供合適的替代方案。
```

Reply:
This entire passage was in written past tense to explain the intentions back in 2015